frontend/src/metabase/query_builder/actions/core/utils.ts

```diff
@@ -69,3 +70,15 @@ export function getAdHocQuestionWithVizSettings(options: {
 
   return question;
 }
+
+export function syncCardParametersWithTemplateTags(question: Question) {
+  const isNewQuestionNative = Lib.queryDisplayInfo(question.query()).isNative;
+
+  if (!isNewQuestionNative) {
+    return question;
+  }
+
+  const parameters = getTemplateTagParametersFromCard(question.card());
+
+  return question.setParameters(parameters);
+}
```

Comment:
This is from the existing diff, just extracted into a new function.

https://github.com/metabase/metabase/pull/50728/files#diff-feedbe46976731ce62d4f714452c04eff7e9d8f6a76ebca1ea2fd1f8864ab780L129-L133